xbmc/pvr/PVRGUIActions.cpp

```diff
@@ -768,19 +823,23 @@ namespace PVR
                                             CVariant{19293}); // "Remove all deleted recordings from trash? This operation cannot be reverted."
   }
 
+  class AsyncUndeleteRecording : public AsyncRecordingAction
```

Comment:
Intention was to have the implementation as close to the using code as possible. But no strong feelings about this here.